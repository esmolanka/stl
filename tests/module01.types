
module Module01

type Either a b =
  | Left : a
  | Right : b

type Maybe a =
  | Nothing
  | Just : a

provide forall a b. Either a b -> Maybe b
