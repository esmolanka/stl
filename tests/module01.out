tests/module01.types:12:16-12:42:
  Normalised:
    ∀ a. ∀ b.
      (∀ (ρ : Row). ∃ (ω : #). ∃ (ω : #). <Left^ω : a| Right^ω/1 : b | ... ρ>)
      -> ∀ (ρ : Row). ∃ (ω : #). ∃ (ω : #).
        <Nothing^ω : Unit| Just^ω/1 : b | ... ρ>
  Kind:
    Type